### PR TITLE
[DOCS] Configuration and component section overhaul

### PR DESCRIPTION
**Pull request by ChrisChinchilla**
This PR does two things… 

1. Overhauls the annotated config files to show all (and there are a lot) potential configuration points and what they mean. There are still some that are unclear, and there are todos in the comments, but it's a start. The reasoning behind this is to provide the higher level content (quoickstarts etc) and these lower level guides as we fill the more valuable gap in the middle slowly.
2. Personally I found the previous content organisation confusing, and it focussed too much on components and not use case. So this PR also reorganises around these config files, moving the components to standardised sections under _components_, with repeating sections for config, API, architecture, or whatever each component needs. Then some of the more "how to" content that was under the older component sections is moved into the how to and operational guide sections, which more helpful naming. Some of the moving of that content is still in progress (i.e., how and why to apply configuration) and this just begins that, it's coming soon.

### CODE REVIEW

**Comment by ChrisChinchilla** [comment]
Probably easiest to look at the preview for review…

**Comment by ChrisChinchilla** [comment]
@nbroyles Should this also be removed?

https://github.com/m3db/m3/blob/31134515468666148886a92d404fea0041f06c75/site/content/includes/m3query/annotated_config.yaml#L64

**Comment by nbroyles** [comment]
> @nbroyles Should this also be removed?
> 
> https://github.com/m3db/m3/blob/31134515468666148886a92d404fea0041f06c75/site/content/includes/m3query/annotated_config.yaml#L64-L67

Yup. You can also remove this: https://github.com/m3db/m3/blob/31134515468666148886a92d404fea0041f06c75/site/content/includes/m3query/annotated_config.yaml#L198-L209

File: site/content/includes/m3aggregator/annotated_config.yaml [hunk 1]
```diff
@@ -0,0 +1,754 @@
...
+# TODO: More detail than this 
+# https://github.com/m3db/m3/blob/9f129cf9f16430cc5a399f60aa5684fb72b55bb5/src/cmd/services/m3query/config/config.go#L116
+# Configuration for logging.
+logging:
+  # Log file location
+  file: <string>
+  # Error logging level
+  level: <string>
+  # TODO: ??
+  fields:
```
Comment (robskillington) [inline site/content/includes/m3aggregator/annotated_config.yaml:11]: Perhaps just remove these?

File: site/content/includes/m3aggregator/annotated_config.yaml [hunk 2]
```diff
@@ -0,0 +1,754 @@
+# Logging configuration
+# TODO: More detail than this 
+# https://github.com/m3db/m3/blob/9f129cf9f16430cc5a399f60aa5684fb72b55bb5/src/cmd/services/m3query/config/config.go#L116
```
Comment (robskillington) [inline site/content/includes/m3aggregator/annotated_config.yaml:3]: Perhaps just remove the TODOs for time being?

File: site/content/includes/m3aggregator/annotated_config.yaml (hunk 1 above)
Comment (ChrisChinchilla) [reply site/content/includes/m3aggregator/annotated_config.yaml:11]: @robskillington I will, but at the moment, there are also placeholders of things to check with you when we review fully ;)

**Comment by ChrisChinchilla** [comment]
Ready for a final review @robskillington and also bringing in @wesleyk here for some input… A couple of notes after many conversations with Rob…

The m3dbnode config is pretty solid and mostly understandable. The M3 Query/Coordinator config isn't, and is still quite confusing, mostly as there seem some discrepancies between the way config works between m3dbnode setups and separated node clusters.

This is something that need resolving in the code base but also docs changes from here on in at a higher level will attempt to clarify the differences in context of implementation.

**Comment by robskillington** [comment]
Overall I think this is looking really good.

I do have one major concern though having reviewed the documentation preview a few times, I think that "Architecture" is now really hard to find. You don't really expect to have "Architecture" under "Reference". 

I believe actually it would be more discoverable/obvious if we did the following:
```
Reference & Config
- M3DB
 - Configuration file
- M3 Coordinator
 - API
 - Configuration file
- M3 Query
 - API
 - Configuration file

Architecture
- M3DB
 - Storage Engine
 - Sharding
 - Consistency Levels
 - Storage
 - Commit Logs And Snapshot Files
 - Peer Streaming
 - Caching
- M3 Coordinator
 - Overview
- M3 Query
 - Overview
```

**Review by robskillington — APPROVED** [review]
LGTM

**Pull request merged**